lib/src/components/progress/progress_circle.dart

```diff
@@ -0,0 +1,131 @@
+import 'dart:math' as math;
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+import '../../../zeta_flutter.dart';
+import 'progress.dart';
+
+/// Sizes for [ZetaProgressCircle]
+enum ZetaCircleSizes {
+  ///24 X 24
+  xs,
+
+  /// 36 X 36
+  s,
+
+  /// 40 x 40
+  m,
+
+  /// 48 X 48
+  l,
+
+  /// 64 X 64
+  xl
+}
+
+///Class definition for [ZetaProgressCircle]
+class ZetaProgressCircle extends ZetaProgress {
+  /// Constructor for [ZetaProgressCircle]
+  const ZetaProgressCircle({
+    super.key,
+    super.progress = 0,
+    this.size = ZetaCircleSizes.xl,
+    this.rounded = true,
+  });
+
+  ///Size of [ZetaProgressCircle]
+  final ZetaCircleSizes size;
+
+  /// Border Type for [ZetaWidgetBorder]
+  final bool rounded;
+
+  @override
+  State<ZetaProgressCircle> createState() => ZetaProgressCircleState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(EnumProperty<ZetaCircleSizes>('size', size))
+      ..add(DoubleProperty('progress', progress))
+      ..add(DiagnosticsProperty<bool>('rounded', rounded));
+  }
+}
+
+///Class definition for [ZetaProgressCircleState]
+class ZetaProgressCircleState extends ZetaProgressState<ZetaProgressCircle> {
+  @override
+  Widget build(BuildContext context) {
+    return AnimatedBuilder(
+      animation: controller,
+      builder: (context, child) {
+        return CustomPaint(
+          size: _getSize(),
+          painter: CirclePainter(
+            progress: animation.value,
+            rounded: widget.rounded,
+          ),
+        );
+      },
+    );
+  }
+
+  Size _getSize() {
+    switch (widget.size) {
+      case ZetaCircleSizes.xs:
+        return const Size(ZetaSpacing.x6, ZetaSpacing.x6);
+      case ZetaCircleSizes.s:
+        return const Size(ZetaSpacing.x9, ZetaSpacing.x9);
+      case ZetaCircleSizes.m:
+        return const Size(ZetaSpacing.x10, ZetaSpacing.x10);
+      case ZetaCircleSizes.l:
+        return const Size(ZetaSpacing.x12, ZetaSpacing.x12);
+      case ZetaCircleSizes.xl:
+        return const Size(ZetaSpacing.x16, ZetaSpacing.x16);
+    }
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DoubleProperty('progress', progress))
+      ..add(DiagnosticsProperty<AnimationController>('controller', controller))
+      ..add(DiagnosticsProperty<Animation<double>>('animation', animation));
+  }
+}
+
+///Class definition for [CirclePainter]
+class CirclePainter extends CustomPainter {
+  ///Constructor for [CirclePainter]
+  CirclePainter({this.progress = 0, this.rounded = true});
+
+  ///Percentage of progress in decimal value, defaults to 0
+  final double progress;
+
+  ///Is circle rounded, defaults to true
+  final bool rounded;
+
+  final _paint = Paint()
+    ..color = Colors.blue
```

Comment:
here we should use ```Zeta.of(context).colors.primary```. You'll need to add a prop for either BuildContext or ZetaColors